### DISCUSSION TITLE
Lesson 11, MintBasicNft: make mint results in error when deploying to sepolia testnet

### DISCUSSION THREAD

**Discussion by corwinthill** [Cyfrin/foundry-full-course-cu#437] [Q&A]
Hey All,

This is my first discussions post, so my formatting and question asking etiquette may be off.
I am having issues when using --> make mint ARGS="--network sepolia"
My Makefile is the same as what is posted on the project github with the addition of the line Patrick wrote at 08:29:50 in the video. I also looked at the other discussions that had problems at this step, so I installed jq and confirmed it installed correctly using jq --version. This probably fixed another issue, but my current one still remained. I messed around with the DevOpsTools.sol and the get_recent_deployment.sh a bit to try and understand them better, but anything I tried did not change my error. After I messed around with those files I deleted the devops package and did a fresh install. 
So now I am asking the community for assistance. I attached an image of my whole error output, as well as the file explorer. The problem seems to be about an issue with a missing file or directory, but the paths displayed in the error all seem to check out. If anyone has any ideas, they would be greatly appreciated.

![Screenshot 2023-07-24 191042](https://github.com/Cyfrin/foundry-full-course-f23/assets/132314343/d3b3417c-3c15-4c52-9abb-ee46bc0b5390)

**Comment by Niferu** [comment]
Hello @corwinthill 

Script just does not see your recently deployed contract to Sepolia network. Make sure you have deployed contract to Sepolia and it's address is correctly gathered by mint script by adding some console.log in your code and running code with -vvvvv

**Reply by corwinthill** [reply]
I followed your suggestion and added lots of console.log into the DevOpsTools.sol file to see if all the parameters were correct, and everything looked right. So I started manually running the get_recent_deployment.sh bash script, and passing in parameters on my own. Running it from the command line returned the correct contract address of my contract deployed on Sepolia. So the bash script seems to be working perfectly fine. This makes me believe that there is something weird going on with the ffi call in the DevOpsTools library. The documentation for ffi in foundry is somewhat lacking so I don't fully understand what it does. 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/132314343/f962ed57-1ad5-4db9-8779-c44d91f7b729)
From my understanding the first two input parameters are telling the ffi call that we want to run a bash script at a certain file location. Then the next 3 parameters are what are being passed into the bash script. This seems like it should work correctly, but it fails for me. When I was running the bash script directly from the command line it did not want commas between my arguments, so I tried changing the input parameters to all be in a single index of the string array with spaces separating them, rather than commas. This did not work either. I am thoroughly stumped now. :/

**Reply by Niferu** [reply]
Can you link your repo here? I will check what might be the issue

**Reply by YOUofAllPeople** [reply]
Running into the same error, installed jq, no change. It feels like a simple folder path issue ("No such file or directory"), it can't seem to find the broadcast folder?

**Reply by scoolj** [reply]
I am also experiencing the same problem. I  think the bashfile has not been able to locate the deployed contract from the broadcast folder.

**Reply by scoolj** [reply]
I also experienced same error.  so I solved with the answer provided by Patrick.

```
sudo apt-get update
sudo apt-get -y install jq
```
After the installation if the problem persist, you may want to check whether the broadcast folder exist on your machine and navigate  to the recently deployed. This will ensure your script is not looking for file that does not exit.

**Comment by mikekochh** [comment]
I had the same issue. Check out this answer to see if this works for you as well: https://github.com/Cyfrin/foundry-full-course-f23/discussions/58
